File: 1940-maximum-xor-for-each-query/1940-maximum-xor-for-each-query.py
class Solution:
    def getMaximumXor(self, nums: List[int], maximumBit: int) -> List[int]:
        prefixXOR=[0]*len(nums)
        prefixXOR[0]=nums[0]
        answer=[]
        for i in range(1,len(nums)):
            prefixXOR[i]=prefixXOR[i-1] ^ nums[i]
        mask=(1<<maximumBit)-1
        for i in range(len(prefixXOR)-1,-1,-1):
            answer.append(prefixXOR[i] ^ mask)
        return answer

        
        
        
            

File: 1940-maximum-xor-for-each-query/1940-maximum-xor-for-each-query.java
class Solution {
    public int[] getMaximumXor(int[] nums, int maximumBit) {
        ArrayList<Integer> prefixXOR=new ArrayList<>();
        prefixXOR.add(nums[0]);
        int[] answer=new int[nums.length];
        for(int i=1;i<nums.length;i++){
            prefixXOR.add(prefixXOR.get(i-1)^nums[i]);
        }
        int mask=(1<<maximumBit)-1;
        for(int i=prefixXOR.size()-1;i>-1;i--){
            answer[nums.length-1-i]=prefixXOR.get(i)^mask;
        }
        return answer;
    }
}